### ISSUE TITLE
Support SQLite's CREATE TABLE ... WITH ROWID

### ISSUE THREAD

**Issue by knutwannheden** [jOOQ/jOOQ#9711]
SQLite's `CREATE TABLE` has a `WITHOUT ROWID` clause and thus also supports tables which don't declare a `_rowid_` column. See https://www.sqlite.org/withoutrowid.html and https://www.sqlite.org/rowidtable.html.

jOOQ should at least support this in the parser.

**Comment by lukaseder** [comment]
I don't think we should support this feature independently of supporting the general idea of clustered indexes: https://github.com/jOOQ/jOOQ/issues/9561

**Comment by JacksonBailey** [comment]
Thank you for the information, I was not aware jOOQ could generate from SQL scripts directly. Unfortunately I still get problems with the h2 version (something about indexes on `CLOB`s which probably came from the `TEXT PRIMARY KEY` in SQLite version). I also went out on a limb and tried using the ignore syntax when using a SQLite database directly but it did not work (unsurprisingly). 

Some more information, jOOQ cannot work on SQLite `WITHOUT ROWID` tables because it assumes the `rowid` column is there and tries to use it. Because of that I don't think it is a good idea to modify the parser to understand `WITHOUT ROWID` tables syntax without changing the semantics.

jOOQ *seems* to be able to work with `STRICT` SQLite tables. In a nutshell, `STRICT` tables have to directly use a "real" column affinity that SQLite supports (instead of the implicit conversions it normally does) and it no longer allows values into the table that could not be coerced. This prevents things like `'a'` from ending up in `INTEGER` columns. What are your thoughts on adding support for parsing SQLite `CREATE TABLE` statements with the `STRICT` table option even if no explicit support is added for them?

**Comment by lukaseder** [comment]
> Thank you for the information, I was not aware jOOQ could generate from SQL scripts directly

I'm sorry, I didn't look at the stack trace thoroughly enough. It arises because the `SQLiteDatabase` needs to parse the table definition for various reasons, including reading check constraints, as those things aren't being made available otherwise in SQLite

> Some more information, jOOQ cannot work on SQLite `WITHOUT ROWID` tables because it assumes the `rowid` column is there and tries to use it. Because of that I don't think it is a good idea to modify the parser to understand `WITHOUT ROWID` tables syntax without changing the semantics.

I don't know what you mean by this, specifically. Perhaps create a new issue with details? It won't be strictly related to supporting the clause in a `CREATE TABLE` statement.

> jOOQ _seems_ to be able to work with `STRICT` SQLite tables. In a nutshell, `STRICT` tables have to directly use a "real" column affinity that SQLite supports (instead of the implicit conversions it normally does) and it no longer allows values into the table that could not be coerced. This prevents things like `'a'` from ending up in `INTEGER` columns. What are your thoughts on adding support for parsing SQLite `CREATE TABLE` statements with the `STRICT` table option even if no explicit support is added for them?

Yeah, jOOQ always pretended SQLite was "strict" and didn't implement this type affinity madness. So, it makes perfect sense to support parsing `STRICT` wherever it is currently happening: I'll implement this right away: https://github.com/jOOQ/jOOQ/issues/14631

**Comment by lukaseder** [comment]
Parser support (ignoring the clause) for `WITHOUT ROWID` is being tracked here: https://github.com/jOOQ/jOOQ/issues/14690

**Comment by lukaseder** [comment]
Having studied this madness, I think we've been doing `CREATE TABLE` wrong all this time, in SQLite, without noticing.

- By default, all tables created should be `WITHOUT ROWID`, when jOOQ creates them to avoid any weird implicit identities. (We're not generating them anyway, given that we produce the `INT` type, not `INTEGER` type, in the absence of an explicit identity)
- Unless there's an identity, in case of which, we already make it explicit

A summary of my updated understanding is here:

- https://github.com/jOOQ/jOOQ/issues/14656#issuecomment-1441859959

**Comment by lukaseder** [comment]
Nope, still got it wrong 😅. While there is a ton of interaction between the `_rowid_` column and the identity column, they *can* be independent.

**Comment by lukaseder** [comment]
Databricks has a similar feature, where we can enable row tracking:

- https://docs.databricks.com/aws/en/delta/row-tracking

It might be interesting to standardise over this behaviour. Oracle has a row movement flag, for example
